documentation/src/test/java/example/SuiteDemo.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package example;
+
+//tag::user_guide[]
+import org.junit.platform.suite.api.SelectPackages;
+import org.junit.platform.suite.api.Suite;
+import org.junit.platform.suite.api.SuiteDisplayName;
+
+@Suite
+@SuiteDisplayName("JUnit Platform Suite Demo")
+@SelectPackages("example")
+//end::user_guide[]
+@org.junit.platform.suite.api.IncludeClassNamePatterns
```

Comment:
Both the platform launcher and platform runner only include classes with specific names when no other pattern is provided. In doing so they ignore `ExampleTestCase.failingTest` which would fail the build. 

I don't think it is a good idea to repeat this feature in the suite engine. It is a "hidden" requirement in addition to the use of the annotations. However I'm also unaware of the rationale for including it in the first place. 

What was is reason to only include classes with certain names by default?